### DISCUSSION TITLE
GitHub Servers Update issue

### DISCUSSION THREAD

**Discussion by toddc01b** [RavenSystem/esp-homekit-devices#1608] [General]
Hi, I have just tried to install firmware 9.0.2 from 8.1 and have now lost all services. I did the update from HAA Home Manager.
In settings properties they are all showing a mDNS status of removed and no longer show up in the mDNS Log
I've carried out updates in the past with no issue. I'm not sure what's gone wrong now.
Any help would be appreciated.

Thanks

**Comment by frazer81** [comment]
I’ve had exactly the same issue.
Would appear all of my devices are stuck in a perpetual upgrade loop according to the HAA logs.
Sign result: error seems to be a common appearance in the log.

Regards,

Sean Frazer
________________________________
From: toddc01b ***@***.***>
Sent: Wednesday, November 3, 2021 7:22:53 PM
To: RavenSystem/esp-homekit-devices ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [RavenSystem/esp-homekit-devices] update 9.0.3 (Discussion #1608)


Hi, I have just tried to install firmware 9.0.3 from 8.1 and have now lost all services. I did the update from HAA Home Manager.
In settings properties they are all showing a mDNS status of removed and no longer show up in the mDNS Log
I've carried out updates in the past with no issue. I'm not sure what's gone wrong now.
Any help would be appreciated.

Thanks

**Reply by toddc01b** [reply]
Hi Sean, 
Have you had any luck fixing this?
After a while that the update loop does seem to stop but the devices still don't work. Cycling power or resetting the device seems to start the loop all over again. 
I have attached a pic of what my error looks like. 
 ![image](https://user-images.githubusercontent.com/69740428/140187246-70f06418-9613-4e9c-beb5-a7f04229f126.jpeg)

**Comment by RavenSystem** [comment]
Error is caused by an issue with SSL/TLS connection with GitHub. You can see things like `! wolfSSL_send 2` in logs. 
GitHub uses a distributed network, and some servers have compatibility issues. Most people updated all devices without any issue.

Using a custom server will solve issue. I will update HAA Installer with better compatibility, but I am not sure about solving that rare issue.

**Comment by RavenSystem** [comment]
Maybe changing DNS Servers and reboot router could force to use different servers.

**Reply by toddc01b** [reply]
I will try the DNS change followed by router re-boot and power cycle my devices tonight to see if that fixes the issue. 

Just checked and my router is set to DNS 1.1.1.2 which is cloudflare. 

Is there a DNS you would suggest using? 1.1.1.1? Or google 8.8.8.8?

Sorry, but I'm still learning. 

Failing the above would you suggest starting over by re-flashing each device?

**Comment by frazer81** [comment]
I had my DNS set to 1.1.1.1 and changed this morning to 8.8.8.8
Power cycled my router and a couple of devices but am still getting the same loop occurring.
Had to get to work so haven't had time to investigate further.
I'll isolate a sonoff on the network so I'm only getting logs from a single device and see what I can find.
It looks like José has already looked at a patch (thank you very much) in v9.0.3 that may address this, though feels like I might need to get the FDTI out and re-flash again.

Keep you posted!
SF

**Comment by Ucoz3000** [comment]
updated 9.0.2 to 9.0.3. Got an endless loop

**Comment by frazer81** [comment]
I’ve done some tests - same problem exists after changing DNS and power cycling everything.

Logs attached - happy to try any other suggestions before I reflash if it helps

Regards,

Sean
________________________________

**Comment by RavenSystem** [comment]
Yesterday, I updated all my devices (about 50) from 9.0.2 to 9.0.3. Half of them updated right, but other half got SSL errors. I rebooted my router, and they updated fine. It is a thing of luck. I use Unbound as DNS server.

Reflashing devices will not help.

Best way to fix this quickly is uploading files to other server. A local web server is a good option. Then, configure each device to use that server.

I am going to do some changes to try to continue process from error instead doing a full reset of the update process.

**Reply by toddc01b** [reply]
I tried the DNS change and router re-boot but the errors were still there. 

With your suggestion below, how do I do this? How do I access each device to configure them? I don't seem to be able to access setup mode anymore. 

"uploading files to other server. A local web server is a good option. Then, configure each device to use that server."

Thanks for all the help so far, I really appreciate it. 

TC

**Comment by RavenSystem** [comment]
To put HAA Installer in setup mode, you must do a power cycle. State changes between setup and normal installation each power cycle.

**Comment by RavenSystem** [comment]
GitHub server shows Unauthorized error when many devices are updating at once:

HTTP/1.1 401 Unauthorized
Connection: keep-alive
Content-Length: 0
x-github-backend: Kubernetes
X-GitHub-Request-Id: 5F7E:2C5A:111d61:96CF62:8381FF0B
Accept-Ranges: bytes
Date: Fri, 05 Nov 2021 09:53:14 GMT
Via: 1.1 varnish
X-Served-By: cache-mad22082-MAD
X-Cache: MISS
X-Cache-Hits: 0
X-Timer: S1636105994.311469,VS0,VE100

**Reply by RavenSystem** [reply]
I am coding a workaround and I hope it will be available soon.

**Comment by RavenSystem** [comment]
I updated HAA Installer to v5.4.0 with a workaround to work with Unauthorized issue.

**Reply by toddc01b** [reply]
What do we need to do at our end to get everything working?

Is the local server option still the solution or does the HAA installer upgrade fix it? If so what's the process?

Thanks again for all of your support. 

TC

**Reply by RavenSystem** [reply]
Let devices to update normally. HAA Installer upgrade will fix it.

**Comment by frazer81** [comment]
I’ve updated this morning - power cycled and let the update commence.
All devices back online and functional!
Thanks Jose!! Really appreciate the updates mate.

Regards,

Sean Frazer
________________________________

**Comment by childledj** [comment]
still no success! I se some files tryin to download but verification sign ends with error. And still looped

UPD> I have use own server for updates and it works!

**Reply by Ucoz3000** [reply]
what is your own update server?

**Reply by childledj** [reply]
I’ve simply create folder via ftp on my website, and upload all files for 9.0.4 into it.
Use port 443, and no “/“ at the end of url

**Reply by toddc01b** [reply]
Does anyone know if the update will work over SMB from a NAS drive on the local network?

**Comment by toddc01b** [comment]
I cycled power and everything has updated and is now back online. 

Thanks so much for all your help.

**Comment by RavenSystem** [comment]
I am going to release a new version of HAA OTA Installer to improve connection with GitHub Servers.

**Reply by fiska1** [reply]
Hello.  I still can't update my devices.  Versions from 9.0 are not updated.  I stitched one device for the new firmware 9.1, but it also does not activate.  What could be the problem?

**Reply by fiska1** [reply]
![image](https://user-images.githubusercontent.com/30092533/141652771-fb9844ad-db97-42e6-9d11-bd2770348a15.png)
